recon_test_pack/simulate_data.sh

```diff
@@ -29,51 +33,71 @@ fi
 if [ $# -gt 4 ]; then
   suffix=$5
 fi
-echo "===  create ACFs"
-calculate_attenuation_coefficients --ACF my_acfs$suffix.hs ${atten_image} ${template_sino} forward_projector_proj_matrix_ray_tracing.par > my_create_acfs${suffix}.log 2>&1
-if [ $? -ne 0 ]; then 
-  echo "ERROR running calculate_attenuation_coefficients. Check my_create_acfs${suffix}.log"; exit 1;
-fi
 
-echo "===  create line integrals"
-forward_project my_line_integrals$suffix.hs  ${emission_image} ${template_sino} forward_projector_proj_matrix_ray_tracing.par > my_create_line_integrals${suffix}.log 2>&1
-if [ $? -ne 0 ]; then 
-  echo "ERROR running forward_project. Check my_create_line_integrals${suffix}.log"; exit 1;
-fi
 
+# Check if the suffix is "_SPECT"
+if [ "$suffix" = "_SPECT" ]; then
 
-echo "=== create constant randoms background"
-${INSTALL_DIR}stir_math -s --including-first \
-         --times-scalar 0 --add-scalar $background_value \
-         my_randoms$suffix my_line_integrals$suffix.hs
-if [ $? -ne 0 ]; then 
-  echo "ERROR running stir_math"; exit 1; 
-fi
+	echo "=== create line integrals for SPECT sinogram"
+	forward_project "my_sino${suffix}.hs" "${emission_image}" "${template_sino}" forward_project_SPECTUB.par > "my_create_sino${suffix}.log" 2>&1
+	if [ $? -ne 0 ]; then 
+			echo "ERROR running forward_project for SPECT sinogram. Check my_create_sino${suffix}.log"; exit 1;
+	fi
 
-echo "===  create norm factors"
-# currently just 1 as not used in rest of script yet.
-stir_math -s --including-first \
-          --times-scalar 0 --add-scalar 1 my_norm$suffix.hs my_acfs$suffix.hs
-
-echo "===  create prompts"
-INPUT=my_line_integrals${suffix}.hs OUTPUT=my_prompts${suffix}.hs \
-     MULT=my_acfs${suffix}.hs \
-     RANDOMS=my_randoms${suffix}.hs \
-     correct_projdata uncorrect_projdata.par > my_create_prompts${suffix}.log 2>&1
-if [ $? -ne 0 ]; then 
-  echo "ERROR running correct_projdata. Check my_create_prompts${suffix}.log"; exit 1;
-fi
 
-# could call poisson_noise here
+	echo "===  create line integrals for attenuation SPECT sinogram"
+	forward_project "my_attenuation_sino$suffix.hs" "${atten_image}" "${template_sino}" > "my_create_attenuation_sino${suffix}.log" 2>&1
+	if [ $? -ne 0 ]; then 
+		echo "ERROR running forward_project for attenuation sinogram. Check my_create_attenuation_sino${suffix}.log"; exit 1;
+  fi
```

Comment:
This is incorrect. The attenuation image is passed in the .par files (both forward project and reconstruction).

It would make no sense to use multiplicative attenuation, as that's the whole point of SRTSPECT (inverse of the attenuated Radon transform).

Reply:
Of course. my apologies.

I think that we need to clarify that in the `SRTSPECTReconstruction` class itself  (and a 1 line comment in the example .par file). Ideally, we also change the keyword to something else then. `attenuation filename` is too ambiguous. I'm not sure what to suggest though. "attenuation projection filename"?

Reply:
Thank you wholeheartedly!!! I think "attenuation projection filename" is nice. 

- Should I leave in the .sh, the comment I suggested above? Or will the comment be removed from there?

-Along with the .par file in the example, shouldn't I also add the same comment in the SRT2DSPECT.h file, where a sample .par file is also included? If I remove the note it will be a 1 line comment.

Essentially, SRT2DSPECT takes two inputs:
 1. The emission sinogram, which represents the measured attenuated data.
 2. The attenuation sinogram, which is the Radon transform (line integrals) 
    of the attenuation map. 
    
Except that this is too long description, it does not have the name of the files; it is more helpful in the .par files if the file names remain in the comments.

I am looking forward to your reply. **THANK YOU WHOLEHEARTEDLY FOR ALL YOUR INVALUABLE HELP!!!**

Reply:
> * Should I leave in the .sh, the comment I suggested above? Or will the comment be removed from there?
> 
> -Along with the .par file in the example, shouldn't I also add the same comment in the SRT2DSPECT.h file, where a sample .par file is also included? If I remove the note from the suggested .sh comment, it will be a 1 line comment.
> 

I think the class definition needs to have all the explanation. I generally try to avoid repetition, as things get out-of-date quickly, but most people prefer to have the doc everywhere. So, choose what you think is best!

> Essentially, SRT2DSPECT takes two inputs:
> 
> 1. The emission sinogram, which represents the measured attenuated data.
> 2. The attenuation sinogram, which is the Radon transform (line integrals)
>    of the attenuation map.
> 
> Except that this is too long description, it does not have the name of the files; it is more helpful in the .par files if the file names remain in the comments. 

This long description is great for the class doc, except that I suggest to write "The attenuation projection sinogram..."

Reply:
thank you wholeheartedly! 
I will remove from .sh. I will write the class comment. 
In the .par file in the examples, should I include comment or not? If yes, I think of the following:

The SRT algorithm receives as input the forward projection of the attenuation image (attensino1.hs)
and the measured emission sinogram (sino1.hs) to reconstruct the emission distribution.

I am looking forward to your reply. **Thank you wholeheartedly!!!**